### ISSUE TITLE
Unsafe floating point optimizations

### ISSUE THREAD

**Issue by billyb2** [bitshifter/glam-rs#217]
Hi, would it be possible to maybe have a feature gate for unsafe floating point optimizations within this crate? The ease of using SIMD vectors is really nice, and I think having the option of having the crate run even faster (at the expense of precision of course) would be a great addition.

**Comment by bitshifter** [comment]
What specific optimizations are you referring to?

**Comment by billyb2** [comment]
Anything that is in std::intrinsics::*_fast

For example:
https://doc.rust-lang.org/std/intrinsics/fn.fadd_fast.html

**Comment by bitshifter** [comment]
Generally with glam I have focussed on functionality that is in stable rust or have a clear path to stabilisation. These intrinsics are unstable and as far as I know there is no plan to stabilise them on the horizon (https://doc.rust-lang.org/beta/unstable-book/library-features/core-intrinsics.html). In general the question of how to do "fast math" in Rust is unanswered (see https://github.com/rust-lang/rust/issues/21690 which has been around since 2015). This is one of the reasons I built glam around explicit SIMD usage.

At some point perhaps Rust will get fast math support, but it might not be via explicit intrinsics, e.g. it could be a function attributes that allows the compiler optimise a block by disabling IEEE compliance, who knows. Therefore this isn't something I plan to spend time on. I would consider a PR but accepting it would depend on what the maintenance burden looked like.
